plugin/pkg/admission/serviceinstancecredentials/lifecycle/admission.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecycle
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	informers "github.com/kubernetes-incubator/service-catalog/pkg/client/informers_generated/internalversion"
+	internalversion "github.com/kubernetes-incubator/service-catalog/pkg/client/listers_generated/servicecatalog/internalversion"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/admission"
+
+	scadmission "github.com/kubernetes-incubator/service-catalog/pkg/apiserver/admission"
+)
+
+const (
+	// PluginName is name of admission plug-in
+	PluginName = "ServiceInstanceCredentialsLifecycle"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(io.Reader) (admission.Interface, error) {
+		return NewCredentialsBlocker()
+	})
+}
+
+// enforceNoNewCredentialsForDeletedInstance is an implementation of admission.Interface.
+// If creating new ServiceInstanceCredentials or updating an existing
+// set of credentials, fail the operation if the ServiceInstance is
+// marked for deletion
+type enforceNoNewCredentialsForDeletedInstance struct {
+	*admission.Handler
+	instanceLister internalversion.ServiceInstanceLister
+}
+
+var _ = scadmission.WantsInternalServiceCatalogInformerFactory(&enforceNoNewCredentialsForDeletedInstance{})
+
+func (b *enforceNoNewCredentialsForDeletedInstance) Admit(a admission.Attributes) error {
+
+	// we need to wait for our caches to warm
+	if !b.WaitForReady() {
+		return admission.NewForbidden(a, fmt.Errorf("not yet ready to handle request"))
+	}
+
+	// We only care about credentials
+	if a.GetResource().Group != servicecatalog.GroupName || a.GetResource().GroupResource() != servicecatalog.Resource("serviceinstancecredentials") {
+		return nil
+	}
+
+	// We don't want to deal with status updates or any other sub resources
+	if a.GetSubresource() != "" {
+		return nil
+	}
+
```

Comment:
One last thing: we should only run this controller when the verb is create

Reply:
We should figure out updates in a follow-up.

Reply:
ok, I had included update on purpose, but I can understand the use case with applying a label.  I've updated the registration of the controller to only fire for Create.